library/Asterisk/ARI/Connector.php

```diff
@@ -0,0 +1,109 @@
+<?php
+namespace Asterisk\ARI;
+
+/**
+ * @brief Wrapper around Application servers ARI requests
+ *
+ * @package Asterisk
+ * @subpackage ARI
+ * @author Ivan Alonso [kaian] <kaian@irontec.com>
+ */
+class Connector
+{
+    /**
+     * Connection information
+     */
+    protected $_host;
+    protected $_port;
+    protected $_user;
+    protected $_pass;
+
+    /**
+     * @var Zend_Log
+     */
+    protected $_logger;
+
+    /**
+     *
+     * Creates a new instance of Asterisk Rest Interface Connector
+     */
+    public function __construct()
+    {
+        $params = array(array('writerName' => 'Null'));
+        $this->_logger = \Zend_Log::factory($params);
+
+        // Get Asterisk credetentials from application configuration
+        $bootstrap = \Zend_Controller_Front::getInstance()->getParam('bootstrap');
+        if (is_null($bootstrap)) {
+            $conf = new \Zend_Config_Ini(APPLICATION_PATH . '/configs/application.ini', APPLICATION_ENV);
+            $config = (Object) $conf->toArray();
+        } else {
+            $config = (Object) $bootstrap->getOptions();
+        }
+
+        $this->_user = $config->ari["userName"];
+        $this->_pass = $config->ari["password"];
+        $this->_port = $config->ari["port"];
+    }
+
+    private function _createConnectionHandler()
+    {
+        $ch = curl_init();
```

Comment:
Umm, seems ok to me. I think this is the first time a HTTP request is done from a mapper so we can change now or later, as you prefer.